### PR TITLE
Clean up history branch ancestor operation

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Clean up history branch ancestor operation

**Why?**
Save copy ancestors

**How did you test it?**
Unit tests

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/nDCHistoryReplicator.go [hunk 1]
```diff
@@ -866,18 +866,18 @@ func (r *nDCHistoryReplicatorImpl) backfillHistory(
...
 	}
-	latestBranchID := historyBranch.GetBranchId()
-	var prevBranchID string
 
-	sortedAncestors := copyAndSortAncestors(historyBranch.GetAncestors())
+	prevTxnID := common.EmptyVersion
+	var lastHistoryBatch *commonpb.DataBlob
+	var prevBranchID string
+	sortAncestors(historyBranch.GetAncestors())
+	sortedAncestors := historyBranch.GetAncestors()
```
Comment (alexshtin) [inline service/history/nDCHistoryReplicator.go:878]: I would add 
```
return ans
```
to `sortAncestors()` and replace this 2 lines with:

```suggestion
	sortedAncestors := sortAncestors(historyBranch.GetAncestors())
```

File: service/history/nDCHistoryReplicator.go [hunk 2]
```diff
@@ -954,9 +956,7 @@ BackfillLoop:
 	return lastEventTime, nil
 }
 
-func copyAndSortAncestors(input []*persistencespb.HistoryBranchRange) []*persistencespb.HistoryBranchRange {
-	ans := make([]*persistencespb.HistoryBranchRange, len(input))
-	copy(ans, input)
+func sortAncestors(ans []*persistencespb.HistoryBranchRange) {
 	if len(ans) > 0 {
```
Comment (alexshtin) [inline service/history/nDCHistoryReplicator.go:960]: Do you really need this check? I think `sort.Slice` should work fine with both nil and empty sliced.

**Review by alexshtin — APPROVED** [review]
Thanks for fixing this!

File: service/history/nDCHistoryReplicator.go (hunk 1 above)
Comment (yux0) [reply service/history/nDCHistoryReplicator.go]: updated

**Pull request merged**